aten/src/ATen/native/cpu/ReduceOpsKernel.cpp

```diff
@@ -291,7 +291,7 @@ static void and_kernel_impl(TensorIterator& iter) {
         iter,
         [=](uint8_t a, uint8_t b) -> uint8_t { return (a && b) ? 1 : 0; },
         [=](Vectorized<uint8_t> a, Vectorized<uint8_t> b) {
-          return a & b;
+          return (a != 0) & (b != 0);
```

Comment:
What does `(a != 0)` return? The SIMD ops often have true => `0b11111111...` instead of `1`, but we definitely want the final values to be `1` or `0` not `255` or `0`. (See the comment below in binary_kernel_reduce_vec).

Reply:
Maybe an extra negate will get us the result we want. -1 => 1, 0 => 0